includes/class-amp-theme-support.php

```diff
@@ -2116,6 +2179,11 @@ static function( Optimizer\Error $error ) {
 				// Indicate the number of validation errors detected at runtime in a query var on the non-AMP page for display in the admin bar.
 				if ( AMP_Validation_Manager::has_cap() ) {
 					$non_amp_url = add_query_arg( AMP_Validation_Manager::VALIDATION_ERRORS_QUERY_VAR, $blocking_error_count, $non_amp_url );
+
+					if ( MobileRedirectManager::is_mobile_request() ) {
+						// Disable mobile redirection to prevent an infinite loop.
+						$non_amp_url = add_query_arg( MobileRedirectManager::NO_AMP_QUERY_VAR, '1', $non_amp_url );
+					}
```

Comment:
I think this is relevant whether it is a mobile request or not:

```suggestion
					// Disable mobile redirection to prevent an infinite loop.
					$non_amp_url = add_query_arg( MobileRedirectManager::NO_AMP_QUERY_VAR, '1', $non_amp_url );
```

Reply:
The query var would have no effect since it is not a mobile request, no?

Reply:
Yes, but only safely so if `should_redirect_via_js` is `false`. It would be safer to add the `noamp` query var always to ensure that JS-based redirection will also not end up doing a redirection.